player-vs-computer.js

```diff
@@ -0,0 +1,240 @@
+  );
+
+  playerTurn();
+
+  // all logic for player turn
+  async function playerTurn() {
+    let guessInput = await ask("Guess!\n\n");
+    let guess = parseInt(guessInput);
+    if (guess === computerNum) {
+      console.log("\nThat's it! You guessed the number!\n");
+      process.exit();
+    } else if (guess < computerNum) {
+      console.log("\nThat's not it. My number is higher.");
+    } else if (guess > computerNum) {
+      console.log("\nThat's not it. My number is lower.");
+    }
+
+    console.log("\nOk, my turn!");
+    computerTurn();
+  };
+
+  // all logic for computer turn
+  async function computerTurn() {
+    let guessResult = await ask(
+      `\nIs your number ${computerGuess}? Yes (Y) or no (N)?\n\n`
+    );
+
+    // if guess is correct, end the game
+    if (guessResult.toUpperCase() === "Y") {
+      console.log(
+        `\nI win! Game over. I guess your code didn\'t account for the fact that computer > ${name}, huh?\n`
+      );
+      process.exit();
+    }
+
+    // if guess is wrong, update range & continue binary search
+    else if (guessResult.toUpperCase() === "N") {
+      updateRange();
+    } else if (guessResult.toUpperCase() !== "Y") {
+      if (guessResult.toUpperCase() !== "N") {
+        console.log(
+          `\nYou didn\'t answer my question... let\'s try that again.`
+        );
+        computerTurn();
+      }
+    };
+
+    async function updateRange() {
+      let rangeGuess = await ask(
+        "\nIs your number higher (H) or lower (L)?\n\n"
+      );
+
+      // if secret number is higher than guess, remove all numbers below guess
+      if (rangeGuess.toUpperCase() === "H") {
+        cheatCheckLow();
+        rangeBottom = computerGuess + 1;
+        let newGuess = binarySearch(rangeBottom, rangeTop);
+        computerGuess = newGuess;
+      }
+
+      // likewise, if number is lower than guess, remove all numbers above guess
+      else if (rangeGuess.toUpperCase() === "L") {
+        cheatCheckHigh();
+        rangeTop = computerGuess - 1;
+        let newGuess = binarySearch(rangeBottom, rangeTop);
+        computerGuess = newGuess;
+      } else if (rangeGuess.toUpperCase() !== "H") {
+        if (rangeGuess.toUpperCase() !== "L") {
+          console.log(`\nYou didn\'t answer my question... let\'s try again.`);
+          await updateRange();
+        };
+      };
+
+      function cheatCheckLow() {
+        if (playerNum < computerGuess) {
+          console.log(
+            `\nThe command line wizard just told me you\'re cheating! Shame on you, ${name}. That\'s game over for me.\n`
+          );
+          process.exit();
+        };
+      };
+
+      function cheatCheckHigh() {
+        if (playerNum > computerGuess) {
+          console.log(
+            `\nThe command line wizard just told me you\'re cheating! Shame on you, ${name} That\'s game over for me.\n`
+          );
+          process.exit();
+        };
+      };
```

Comment:
These two could be combined to create one function as well. Having an if statement for if it's higher, and an else if for if it's lower would make your code neater